### PR TITLE
OPENEUROPA-776: Make sure component dependencies are as relaxed as possible

### PR DESCRIPTION
**Pull request by aritomelo**
Update consolidation/robo package from 1.2.2 to 1.2.0 with caret so it will sticks closer to semantic versioning, and allow non-breaking updates.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -33,3 +33,7 @@ matrix:
   IMAGE_PHP:
     - fpfis/httpd-php-dev:5.6
     - fpfis/httpd-php-dev:7.1
+
+  COMPOSER_BOUNDARY:
```
Comment (pfrenssen) [inline .drone.yml]: The matrix is extended with choices for the composer boundary, but this `${COMPOSER_BOUNDARY}` variable is not used in the pipeline, so it has no effect.

For an example on how to use this, see the pipeline of the Code Review component: https://github.com/openeuropa/code-review/blob/master/.drone.yml#L12

**Comment by pfrenssen** [comment]
The `require-dev` section contains the following line:

```
"phpunit/phpunit": "^5||^6.0"
```

Let's also fix this now, it should not have two pipes (should have `|` instead of `||`) and the version numbers are inconsistent with the rest of the file (and even inconsistent within itself).

Let's change it to

```
"phpunit/phpunit": "~5.0|~6.0"
```

**Comment by pfrenssen** [comment]
The CI pipeline is now fixed and is correctly installing the lowest dependencies.

The failure in PHPUnit is because our code is in fact not compatible with PHPUnit 5.0. We are depending on the new namespaced asserts which were introduced in PHPUnit 5.5.0.

The coding standards violations are less clear. The related dependencies are the following:

Dependency | Lowest | Highest
-- | -- | --
openeuropa/code-review | 0.3.0 | 0.3.0
phpro/grumphp | 0.12.0 | 0.14.2
phpmd/phpmd | 2.6.0 | 2.6.0
drupal/coder | 8.2.0 | 8.2.12
squizlabs/php_codesniffer | 2.1.0 | 2.9.1

It is not clear which is responsible for the violations. As for this component we only depend directly on `openeuropa/code-review` so we should avoid adding any of the other ones. I see that we have a 1.0.0-alpha1 release for `openeuropa/code-review` so I would suggest to update our dependency to `~1.0` and see if this will improve matters.

**Comment by aritomelo** [comment]
Testing that suggestion, let's see how it behaves.

File: bin/run [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-    $autoloaderPath = __DIR__.'/../vendor/autoload.php';
-} elseif (file_exists(__DIR__.'/../../../autoload.php')) {
-    $autoloaderPath = __DIR__ . '/../../../autoload.php';
-} else {
-    die("Could not find autoloader. Run 'composer install'.");
-}
-$classLoader = require $autoloaderPath;
-
-$runner = new TaskRunner(new ArgvInput(), new ConsoleOutput(), $classLoader);
-exit($runner->run());
```
Comment (ademarco) [inline bin/run]: @aritomelo this should we put back, it's the actual task Runner executable.

**Comment by voidtek** [comment]
I put again the /bin/run and I changed the composer.json to have all test green.

**Comment by Medi4tor** [comment]
@pfrenssen you to say if the rest is OK, please.

**Pull request merged**